starboard/nplb/posix_compliance/posix_epoll_test.cc

```diff
@@ -0,0 +1,272 @@
+  // First remove should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0);
+}
+
+TEST_F(PosixEpollTest, RainyDayDoubleAddFails) {
+  // First add should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Second add should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0);
+
+  // Remove should succeed, to clear epoll queue for next test.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0)
+      << strerror(errno);
+}
+
+TEST_F(PosixEpollTest, RainyDayDoubleRemoveFails) {
+  // First add should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Remove should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Remove after remove should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0);
+}
+
+TEST_F(PosixEpollTest, RainyDayAddRemoveRepeated) {
+  // First remove should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0);
+
+  // First add should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Second add should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0);
+
+  // Remove should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Remove after remove should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0);
+
+  // Add after remove should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Second add after remove should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0);
+
+  // Remove should succeed, to clear epoll queue for next test.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0)
+      << strerror(errno);
+}
+
+TEST_F(PosixEpollTest, RainyDayInvalidSocket) {
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_ADD, -1, &epev), 0);
+}
+
+TEST_F(PosixEpollTest, RainyDayNoInterest) {
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, nullptr), 0);
+}
+
+TEST_F(PosixEpollTest, SunnyDayWaitTimeout) {
+  // Create listen socket, bind and listen on <port>.
+  int listen_sock = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+  ASSERT_GE(listen_sock, 0);
+
+  // Set socket to be reusable.
+  const int on = 1;
+  int result =
+      setsockopt(listen_sock, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on));
+  EXPECT_EQ(result, 0) << strerror(errno);
+
+  // Bind socket with local address.
+  result =
+      bind(listen_sock, reinterpret_cast<struct sockaddr*>(&address), add_len);
+  EXPECT_EQ(result, 0) << strerror(errno);
+
+  result = listen(listen_sock, kMaxConn);
+  EXPECT_EQ(result, 0) << strerror(errno);
+
+  struct epoll_event ev, events[kMaxEvents];
+  ev.events = EPOLLIN;
+  ev.data.fd = listen_sock;
```

Comment:
Prefer designated initializers when there's no unions:
```
  struct epoll_event event = {
      .events = EPOLLIN,
      .data = {
        .fd = listen_sock,
      }
  };
```
which has the added value of initializing to their default values (i.e. 0) 
all the other unspecified variables (whereas the present code leaves them
random).

Also consider `std::array<struct epoll_event, kMaxEvents> events` or
`std::vector<struct epoll_event> events(kMaxEvents);` if `std::array<>` 
complains. In l.189, use `events.data()` either way.